feat: add Favicons to cheatsheet

Boettner-eric opened the pull request:
I wanted to add some icons for URLs and decided to just try to fetch the favicons from google. This implementation just adds a new FavIcon that respects the existing `showAppIcons` setting and shows FavIcons for any urls in the cheatsheet.

It might be a bad idea to fetch these on each run but I couldn't come up with a simple way to save icons with the json config. I'm open to feedback here if that is a better solution.

I have never written swift before so let me know if there are any clear improvements I can make to this code.

mikker commented:
TIL Google has a favicon service.

I like the idea even though I share your slight worry about fetching icons all the time. Does `AsyncImage` come with a cache option perhaps?

ChatGPT's answer:

> Yes, AsyncImage in Swift does use a cache. It leverages URLSession and the system-wide URL cache (URLCache). However, this caching is limited and managed by the system.
> If you need more control over caching, consider using a third-party library like Kingfisher or SDWebImageSwiftUI, or implement a custom URLCache.

Boettner-eric commented:
Ya I ran into it in an earlier project and have found it pretty useful. 

I'll look into some cacheing via kingfisher or some other package, that seems like a necessary feature and it looks like the default AsyncImage caching isn't working the way I would expect.

Boettner-eric commented:
@mikker I updated this to include use Kingfisher for the image caching. I tested locally and it seems to properly cache but it might be worth it to get another opinion. My xcode also added a swift version and a swift-syntax package that I can't seem to remove 🤷

mikker commented:
Closed in #177. Great work!
